app/routes/submissions.rb

```diff
@@ -15,8 +15,15 @@ class Submissions < Core
 
         title("%s by %s in %s" % [submission.problem.name, submission.user.username, submission.problem.language])
 
+        comment_history = if current_user.guest? || !$flipper[:comment_history].enabled?(current_user)
+          []
+        else
+          current_user.comments.recent(25)
```

Comment:
Every time an user checks a submission a query like this will be triggered, let's keep an eye on performance issues.